athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/HbaseTableNameUtils.java

```diff
@@ -0,0 +1,152 @@
+     *
+     * @param tableName An Athena Federation SDK TableName.
+     * @return The corresponding HBase table name string.
+     */
+    public static String getQualifiedTableName(TableName tableName)
+    {
+        return getQualifiedTableName(tableName.getSchemaName(), tableName.getTableName());
+    }
+
+    /**
+     * Helper which goes from a schema and table name to an HBase TableName
+     * @param schema the schema name
+     * @param table the name of the table
+     * @return The corresponding HBase TableName
+     */
+    public static org.apache.hadoop.hbase.TableName getQualifiedTable(String schema, String table)
+    {
+        return org.apache.hadoop.hbase.TableName.valueOf(getQualifiedTableName(schema, table));
+    }
+
+    /**
+     * Helper which goes from an Athena Federation SDK TableName to an HBase TableName.
+     *
+     * @param tableName An Athena Federation SDK TableName.
+     * @return The corresponding HBase TableName.
+     */
+    public static org.apache.hadoop.hbase.TableName getQualifiedTable(TableName tableName)
+    {
+        return org.apache.hadoop.hbase.TableName.valueOf(getQualifiedTableName(tableName));
+    }
+
+    /**
+     * Gets the hbase table name from Athena table name. This is to allow athena to query uppercase table names
+     * (since athena does not support them). If an hbase table name is found with the athena table name, it is returned.
+     * Otherwise, tryCaseInsensitiveSearch is used to find the corresponding hbase table.
+     *
+     * @param tableName the case insensitive table name
+     * @return the hbase table name
+     */
+    public static org.apache.hadoop.hbase.TableName getHbaseTableName(HBaseConnection conn, TableName athTableName)
+            throws IOException
+    {
+        if (conn.tableExists(getQualifiedTable(athTableName))) {
+            return getQualifiedTable(athTableName);
+        }
+        Optional<org.apache.hadoop.hbase.TableName> caseInsensitiveMatch = tryCaseInsensitiveSearch(conn, athTableName);
+        if (caseInsensitiveMatch.isPresent()) {
+            return caseInsensitiveMatch.get();
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * Performs a case insensitive table search by listing all table names in the schema (namespace), mapping them
+     * to their lowercase transformation, and then mapping the given tableName back to a unique table. To prevent ambiguity,
+     * an IllegalStateException is thrown if multiple tables map to the given tableName.
+     * @param conn the HBaseConnection used to retrieve the tables
+     * @param tableName The Athena TableName to find the mapping to
+     * @return The HBase TableName containing the found HBase table and the Athena Schema (namespace)
+     * @throws IOException 
+     */
+    @VisibleForTesting
+    protected static Optional<org.apache.hadoop.hbase.TableName> tryCaseInsensitiveSearch(HBaseConnection conn, TableName tableName)
```

Comment:
I would add for the reader of this function is that HBase actually supports case sensitive tables; meaning two tables with the same name can co-exist. And we are attempting to resolve case insensitive `only and only if` there is one table per that string.

Reply:
I think that is covered with "To prevent ambiguity, an IllegalStateException is thrown if multiple tables map to the given tableName."